src/hooks/useUserAccessConfig.js

```diff
@@ -0,0 +1,212 @@
+import { useState, useCallback, useMemo } from "react";
+
+import * as Yup from "yup";
+
+import { axiosPrivate } from "~/_api";
+import config from "~/constants/endpoints.json";
+import { useGlobalSetting } from "~/contexts/GlobalSettingProvider";
+
+const currentConfig = import.meta.env.MODE === "development" ? config.test : config.prod;
+
+const pageConfigInitial = {
+  name: "",
+  email: "",
+  phone_number: "",
+  access: [],
+  orderNumber: 0
+};
+const useUserAccessConfig = () => {
+  const [formData, setFormData] = useState(pageConfigInitial);
+  const [errors, setErrors] = useState({});
+  const [activeTab, setActiveTab] = useState(0);
+  const { setting } = useGlobalSetting();
+  const [modalOpenAdd, setModalOpenAdd] = useState(false);
+  const { labels } = setting;
+  const [pageDialogTitle, setPageDialogTitle] = useState("");
+
+  const schema = Yup.object().shape({
+    name: Yup.string().required("Name is required"),
+    email: Yup.string().required("Email is required "),
```

Comment:
This will be good If you put it out side of component ,it prevents recreation , on my part I fixed